File: app/src/main/java/com/example/epigram/data/EpigramService.java
package com.example.epigram.data;

import androidx.annotation.Nullable;

import com.example.epigram.data.model.PostTemplate;
import com.example.epigram.data.model.SearchResult;
import com.example.epigram.data.model.Wrapper;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface EpigramService {

    @GET("ghost/api/v2/content/posts")
    Single<Wrapper<PostTemplate>> getPostsFilter(@Query("key") String key, @Query("include") String include, @Nullable @Query("filter") String fiter, @Query("limit") String limit, @Query("page") int page, @Query("order") String order);

    @GET("ghost/api/v2/content/posts")
    Single<Wrapper<PostTemplate>> getPostsBreak(@Query("key") String key, @Query("include") String include, @Nullable @Query("filter") String fiter, @Query("limit") String limit, @Query("order") String order);

    @GET("ghost/api/v2/content/posts")
    Single<Wrapper<SearchResult>> getSearchIDs(@Query("key") String key, @Query("include") String include, @Query("limit") String theLimit, @Nullable @Query("fields") String fields, @Query("page") int page, @Query("order") String order);

    @GET("ghost/api/v2/content/posts/{id}")
    Single<Wrapper<PostTemplate>> getPostFromNotification(@Path("id") String id, @Query("key") String key, @Query("include") String include);

}

File: app/src/main/java/com/epigram/android/data/api/EpigramService.kt
package com.epigram.android.data

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface EpigramService {

    @GET("ghost/api/v2/content/posts")
    fun getPostsFilter(@Query("key") key: String, @Query("include") include: String, @Query("filter") fiter: String?, @Query("limit") limit: String, @Query("page") page: Int, @Query("order") order: String): Single<Wrapper<PostTemplate>>

    @GET("ghost/api/v2/content/posts")
    fun getPostsBreak(@Query("key") key: String, @Query("include") include: String, @Query("filter") fiter: String?, @Query("limit") limit: String, @Query("order") order: String): Single<Wrapper<PostTemplate>>

    @GET("ghost/api/v2/content/posts")
    fun getSearchIDs(@Query("key") key: String, @Query("include") include: String, @Query("limit") theLimit: String, @Query("fields") fields: String?, @Query("page") page: Int, @Query("order") order: String): Single<Wrapper<SearchResult>>

    @GET("ghost/api/v2/content/posts/{id}")
    fun getPostFromNotification(@Path("id") id: String, @Query("key") key: String, @Query("include") include: String): Single<Wrapper<PostTemplate>>

}
